snuba/settings/validation.py

```diff
@@ -95,3 +95,14 @@ def validate_settings(locals: Mapping[str, Any]) -> None:
         assert (
             physical_part < locals["LOCAL_SLICES"]
         ), f"physical slice for logical partition {logical_part} is {physical_part}, but only {partition_count} physical slices exist"
+
+    # build a set of sliced Kafka logical topics
+    sliced_topics = set()
+    for topic_tuple in locals["SLICED_KAFKA_TOPICS"]:
+        sliced_topics.add(topic_tuple[0])
+
+    for logical_topic_name in sliced_topics:
+        for slice_num in range(0, locals["LOCAL_SLICES"]):
```

Comment:
I may have missed something, but isn't the slices number a property of a dataset or storage? Why do we have a global `LOCAL_SLICES` setting ?

Reply:
I'm thinking through what the level of effort would be to change this to be per-storage. For future extensibility it may be worth more complexity

Reply:
FWIW I do think `LOCAL_SLICES` belongs in settings so that it is deployment-specific and not polluting the dataset configuration

Reply:
Updated PR (https://github.com/getsentry/snuba/pull/3226) addresses the removal of `LOCAL_SLICES` and the change to make slice number a property of storage